lib/interface_quda.cpp

```diff
@@ -5424,6 +5430,464 @@ void performGFlowQuda(void **h_out, void **h_in, QudaInvertParam *inv_param, Qud
+          GFlowStep(g_W2, gaugeTemp, g_W1, smear_param->epsilon, smear_param->smear_type, WFLOW_STEP_W2);
+          GFlowStep(g_VT, gaugeTemp, g_W2, smear_param->epsilon, smear_param->smear_type, WFLOW_STEP_VT);
+
+      }
+    // init auxilliary fields [0], [1] and [2] as [3]
+    f_temp0 = f_temp3;
+    f_temp1 = f_temp3;
+    f_temp2 = f_temp3;
+
+    // [4] = Lap2 [0]
+    copyExtendedGauge(precise, g_W2, QUDA_CUDA_FIELD_LOCATION);
+    precise.exchangeGhost();
+    ApplyLaplace(f_temp4, f_temp0, precise, 4, a, b, f_temp0, parity, comm_dim, profile);  
+
+    // [4] -> 3/4 eps [4]  
+    blas::ax(smear_param->epsilon * 3. / 4., f_temp4);
+    
+    // [2] = [4]
+    f_temp2 = f_temp4;
+    
+    // [4] = Lap1 [2]
+    copyExtendedGauge(precise, g_W1, QUDA_CUDA_FIELD_LOCATION);
+    precise.exchangeGhost();  
+    ApplyLaplace(f_temp4, f_temp2, precise, 4, a, b, f_temp2, parity, comm_dim, profile); 
+    
+    // [3] -> [3] + 8/9 eps [4]  
+    blas::axpy(smear_param->epsilon * 8. / 9., f_temp4, f_temp3);
+    
+    // [1], [4] <- [3]
+    f_temp1 = f_temp3;
+    f_temp4 = f_temp1;
+    
+    // [4] <- [4] - 8/9 [2]
+    blas::axpy(-8. / 9.,f_temp2, f_temp4);
+    
+    // [0] <- Lap0 [4]
+    copyExtendedGauge(precise, g_W0, QUDA_CUDA_FIELD_LOCATION);
+    precise.exchangeGhost();  
+    ApplyLaplace(f_temp0, f_temp4, precise, 4, a, b, f_temp4, parity, comm_dim, profile); 
+    
+    // [0] <- 1/4 eps [0]; [0] <- [2] + [0]; [0] <- [1] + [0]
+    blas::ax(smear_param->epsilon * 1. / 4., f_temp0);
+    blas::axpy(1.,f_temp2, f_temp0);
+    blas::axpy(1.,f_temp1, f_temp0);
+    
+    // fout = f_temp0;
+    //redefining f_temp0 to restart loop
+    f_temp3 = f_temp0;
+    
+    i_glob++;
+  }
+
+}    
+    
+/* total_dist == n_steps, n_b is dividing factor of each block, n_Save is the size of the list, "front" denotes whether split hierarchy goes to existing or new subhierarchy */
+std::vector<int> get_hier_list(int total_dist, int n_b, int n_save, bool front = true){
+    
+    std::vector<int> hier_list;
+    int counter = 0;
+
+    int val = total_dist;
+    for (int i_s = 0; i_s < n_save; i_s++) {
+        val = (val <= 1) ? 1 : val / n_b;
+        hier_list.push_back(val);
+        counter += val;
+    }
+    
+    if (front) hier_list.at(0) += total_dist - counter;
+    else hier_list.back() += total_dist - counter;
+  
+    return hier_list;
+    
+}
+    
+int modify_hier_list(std::vector<int> &hier_list, int n_b, int n_save, int threshold) {
+    
+    int result = -1;
+    int current_size = hier_list.size();
+    std::vector<int> temp_list;
+    if (current_size > n_save) errorQuda("something isnt right\n");
+    
+    int diff = n_save - current_size;
+    
+    
+    for (int i=current_size - 1; i >= 0; --i){
+
+        if (hier_list[i] > threshold){
+            
+            temp_list = get_hier_list(hier_list[i], n_b, diff+1,false);
+            // for (int ii = 0; ii< temp_list.size();ii++) printf("tempf %d \n",temp_list[ii]);
```

Comment:
If you want to keep debug prints like this, best to put them inside a boolean on debug verbosity